### ISSUE TITLE
`Out of range value` error when merging agents

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3871]
http://localhost/specify/overlay/merge/Agent/?records=44951%2C44082%2C47102%2C47101

`herb_rbge` database, `auditlog-agent-merging-optimization` and `v7.9-dev`

```
DataError at /api/specify/agent/replace/44082/ (1264, "Out of range value for column 'newrecordid' at row 1") 
```

Happens immediately upon pressing **Merge**

<img width="1417" alt="image" src="https://github.com/specify/specify7/assets/37256050/52c13ffa-d2cc-4261-ac71-f98ad1054ad4">

[Specify 7 Crash Report - 2023-07-31T17 58 13.158Z.txt](https://github.com/specify/specify7/files/12217864/Specify.7.Crash.Report.-.2023-07-31T17.58.13.158Z.txt)

**Comment by grantfitzsimmons** [comment]
http://localhost/specify/overlay/merge/Agent/?records=47117%2C47116%2C47115%2C47114%2C47113%2C47112%2C47111%2C47110%2C47109%2C47108%2C47107%2C47106%2C47105

<img width="938" alt="image" src="https://github.com/specify/specify7/assets/37256050/2ab79c63-1da5-47f4-acba-6d16b957083c">

Same database, `herb_rbge`

[Specify 7 Crash Report - 2023-07-31T19 07 56.535Z.txt](https://github.com/specify/specify7/files/12220076/Specify.7.Crash.Report.-.2023-07-31T19.07.56.535Z.txt)

**Comment by acwhite211** [comment]
This issue is resolved by changing the `newrecordid` column in `spmerge` table from `SmallIntegerField` to `IntegerField` in the PR https://github.com/specify/specify7/pull/3855

**Comment by grantfitzsimmons** [comment]
@melton-jason says this has been fixed in #3855
